mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ContainerEventHandlerServiceConfigurator.java

```diff
@@ -35,45 +37,54 @@
 import org.wildfly.clustering.service.AsyncServiceConfigurator;
 import org.wildfly.clustering.service.FunctionalService;
 import org.wildfly.clustering.service.ServiceConfigurator;
-import org.wildfly.clustering.service.SimpleServiceNameProvider;
 
 /**
  * Service starting mod_cluster service.
  *
  * @author Jean-Frederic Clere
  * @author Radoslav Husar
  */
-public class ContainerEventHandlerServiceConfigurator extends SimpleServiceNameProvider implements ServiceConfigurator, Supplier<ModClusterService>, Consumer<ModClusterService> {
+public class ContainerEventHandlerServiceConfigurator implements ServiceConfigurator, Supplier<ModClusterService>, Consumer<ModClusterService> {
 
-    public static final ServiceName SERVICE_NAME = ModClusterConfigResourceDefinition.MOD_CLUSTER_CAPABILITY.getCapabilityServiceName();
-    public static final ServiceName CONFIG_SERVICE_NAME = SERVICE_NAME.append("config");
+    private final String proxyName;
+    private final LoadBalanceFactorProvider factorProvider;
+    private volatile Supplier<ModClusterConfiguration> configuration;
```

Comment:
This can be final - you can create this object in the constructor.
e.g.
private final SupplierDependency<ModClusterConfiguration> configuration;

ContainerEventHandlerServiceConfigurator(PathAddress address, LoadBalanceFactorProvider factorProvider) {
   super(ProxyConfigurationResourceDefinition.Capability.SERVICE, address);
   this.configuration = new ServiceSupplierDependency<>(new ProxyConfigurationServiceConfigurator(address));
}